### PR TITLE
feat(security): reduce permissions

### LINKED ISSUE #25: ml-workspace: Disable SSH access
To the extent of my experimentation, SSH (sshd and sslh through supervisor) could not be disabled on ml-workspace while still allowing connections through Kubeflow (only localhost/port-forward would work). I'm not fully sure what's going on under the hood here. This may be already covered by a security policy on another layer, but we need to make sure that external SSH connections (or at least the data leak avenues they can enable:  scp, sftp, rsync, sshfs...) cannot be made.

### LINKED ISSUE #44: ml-workspace: Double-check/reduce permissions
Further trim down excessive chmods and chowns, or confirm what what seems to be necessary. #34

### PR DESCRIPTION
**Pull request by frazs**
Resolves https://github.com/StatCan/kubeflow-containers/issues/44
Resolves https://github.com/StatCan/kubeflow-containers/issues/25

I did not find a non-root approach for cron, netdata, and rsyslogd. This may still be possible, but is complicated, so if it is a priority it warrants separate issues. I changed the entry in the sudoers file to be specific to the supervisor startup for these programs.

The sha256sum for vs-code-desktop has changed: the install gets the latest version, and there has been an update since the last time that layer was built.

### CODE REVIEW

**Review by brendangadd — APPROVED** [review]

**Comment by brendangadd** [comment]
@frazs I approved, but I'm wondering if `jovyan` actually needs `sudo` on cron etc. Shouldn't the init system take care of launching and monitoring these processes automatically? (If it does, then I do like the specificity of the grants.)

There are some merge conflicts to be resolved. PR needs a rebase.

**Comment by frazs** [comment]
**Do not merge yet:** while I have rebased and resolved the merge conflicts, I was not able to complete testing. Build fails for an unrelated reason: the ESlint and Markdown VS Code extensions are no longer at their given location. I need to track down replacements or remove them, and then sort out anything that may or may not come up on further layers.

**Comment by frazs** [comment]
**OK to merge pull request**: The VS code extension issue turned out to be a temporary outage, and there are no further issues. I have successfully built and confirmed the functionality of all 3 images.

**Pull request merged**
